### ISSUE TITLE
Add OWNERS files in different subsystems

### ISSUE THREAD

**Issue by kargakis** [openshift/origin#9273]
Already done upstream. This also has come up in discussions with @legionus 

@liggitt @deads2k @smarterclayton @ncdc @bparees @mfojtik @pweil-  @danmcp

**Comment by kargakis** [comment]
@soltysh @miminar @ironcladlou @derekwaynecarr @pmorie @knobunc

**Comment by ncdc** [comment]
+1, and also start assigning (automatically via bot?) PRs to the appropriate people for review

**Comment by derekwaynecarr** [comment]
+1 to both owners and assignee

**Comment by deads2k** [comment]
Are we having trouble getting the right reviewers?  I'm not opposed, but I wouldn't want it to be a solution in search of problem.  Everyone with merge powers can assign their pull already, right?

**Comment by kargakis** [comment]
@legonius please remind me why you wanted this

**Comment by smarterclayton** [comment]
OWNERS is useful for finding people to nag.  Especially for new users.

**Comment by soltysh** [comment]
+1 on the idea itself and I guess it will be in @stevekuznetsov area :D

**Comment by stevekuznetsov** [comment]
Since we can do multiple assignments now, do we want to just assign anyone that's an owner of a directory that the PR touches a file under?

**Comment by legionus** [comment]
I think yes.

**Comment by eparis** [comment]
I always feel like group assignment means that no one has responsibility...

**Comment by eparis** [comment]
I'll just believe @pweil- is gonig to hanve this issue and he's going to think that I will....

**Comment by stevekuznetsov** [comment]
Before we implement anything, first we need to have a conversation on what it means to assign someone a PR. In Origin today we don't even do that. Then I'd like to see a clear set of gains for an OWNERS impl in general and some consensus on what that sort of impl would look like.

**Comment by soltysh** [comment]
I'd assume it looks similar to the one k8s has. Iow. from the list of OWNERS for dir one is picked randomly and he gets the assignment. In general I think it's all about having someone getting the notification that there's such issue, so he can either work on it or re-assign to more appropriate person if such exists. That, I believe, is how it works for k8s, but @eparis will know better.

**Comment by eparis** [comment]
that is what we do, yes. The bot reads the OWNERS files in master, does a little heuristic to weigh people and then from the weight randomly picks the a person and assigns the PR to that person.

**Comment by stevekuznetsov** [comment]
Since that workflow currently lives under the purview of @danmcp  -- two questions:
 - is the way @danmcp is doing it now not sufficient?  how so? how can this approach remedy that?
 - @danmcp would you rather not do this yourself?

**Comment by eparis** [comment]
The OWNERS stuff upstream only touches PRs. And I know @danmcp looks at both PRs and issues. (we have minimal issue management, if you create a PR that fixes and issue, the issue gets assigned to you)

**Comment by danmcp** [comment]
@stevekuznetsov I actually haven't been assigning PRs for review all that much.  We haven't actually had a high enough number of external PRs that we have needed an automated system either.  I did happen to go through all the PRs today and found 5 or 6 of a few hundred PRs that were submitted externally that hadn't gotten enough attention in the last few months.  It would have been useful to auto assign them.  But I also gave myself a todo of going over these every day like I do issues from now on.  I don't have any issue with automating the process a little certainly.  But it's not a huge win right now given the incoming rate.
